packages/@aws-cdk/integ-tests/README.md

```diff
@@ -116,9 +151,206 @@ const testCase = new IntegTestCase(stack, 'CustomizedDeploymentWorkflow', {
     },
   },
 });
+```
+
+### IntegTestCaseStack
+
+In the majority of cases an integration test will contain a single `IntegTestCase`.
+By default when you create an `IntegTest` an `IntegTestCase` is created for you
+and all of your test cases are registered to this `IntegTestCase`. The `IntegTestCase`
+and `IntegTestCaseStack` constructs are only needed when it is necessary to
+defined different options for individual test cases.
+
+For example, you might want to have one test case where `diffAssets` is enabled.
+
+```ts
+declare const app: App;
+declare const stackUnderTest: Stack;
+const testCaseWithAssets = new IntegTestCaseStack(app, 'TestCaseAssets', {
+  diffAssets: true,
+});
+
+new IntegTest(app, 'Integ', { testCases: [stackUnderTest, testCaseWithAssets] });
+```
+
+## Assertions
+
+This library also provides a utility to make assertions against the infrastructure that the integration test deploys.
+
+The easiest way to do this is to create a `TestCase` and then access the `DeployAssert` that is automatically created.
+
+```ts
+declare const app: App;
+declare const stack: Stack;
+
+const integ = new IntegTest(app, 'Integ', { testCases: [stack] });
+integ.assert.awsApiCall('S3', 'getObject');
+```
+
+### DeployAssert
+
+Assertions are created by using the `DeployAssert` construct. This construct creates it's own `Stack` separate from
+any stacks that you create as part of your integration tests. This `Stack` is treated differently from other stacks
+by the `integ-runner` tool. For example, this stack will not be diffed by the `integ-runner`.
+
+Any assertions that you create should be created in the scope of `DeployAssert`. For example,
+
+```ts
+declare const app: App;
+
+const assert = new DeployAssert(app);
+new SdkQuery(assert, 'GetObject', {
+  service: 'S3',
+  api: 'getObject',
+});
+```
+
+`DeployAssert` also provides utilities to register your own assertions.
+
+```ts
+declare const myCustomResource: CustomResource;
+declare const app: App;
+const assert = new DeployAssert(app);
+assert.strictEquals(
+  'CustomAssertion',
+  ExpectedResult.fromObject({ foo: 'bar' }),
+  ActualResult.fromCustomResource(myCustomResource, 'data'),
+);
+```
+
+In the above example an assertion is created that will trigger a user defined `CustomResource`
+and assert that the `data` attribute is equal to `{ foo: 'bar' }`.
+
+### SdkQuery
+
+A common method to retrieve the "actual" results to compare with what is expected is to make an
+AWS API call to receive some data. This library does this by utilizing CloudFormation custom resources
+which means that CloudFormation will call out to a Lambda Function which will
+use the AWS JavaScript SDK to make the API call.
+
+This can be done by using the class directory:
+
+```ts
+declare const assert: DeployAssert;
+
+new SdkQuery(assert, 'MyAssertion', {
```

Comment:
Is this class name still accurate? Shouldn't it match the method name?

Reply:
Good point, I'll rename.